src/modules/Dashboard/features/RespondentData/RespondentDataReview/RespondentDataReview.tsx

```diff
@@ -144,11 +144,13 @@ export const RespondentDataReview = () => {
     const datesResult = data?.result;
     if (!datesResult) return;
 
-    const submitDates = datesResult.dates.map(parseDateToMidnightUTC);
+    const submitDates = datesResult.dates.map(parseDateToMidnightLocal);
     setResponseDates(submitDates);
   });
 
   const handleGetSubmitDates = (date: Date) => {
+    const activityOrFlowId = activityId || activityFlowId;
```

Comment:
Nit: the suggested change below is not going to affect behaviour of this screen, for now (or maybe ever). However, checking for the presence of a flow ID before activity ID is better aligned with the hypothetical possibility that you could be in a context where you're looking at an activity flow _and_ an activity within that flow. Our routes don't support that yet and we haven't built that out in the context of Dataviz – but one day, we might. So it's safer to check if there's a flow present before assuming we're only looking at an activity.  (This order of checks is done more commonly in the backend and respondent apps, but I figured it wouldn't hurt to do the same here.)

```suggestion
    const activityOrFlowId = activityFlowId || activityId;
```

Reply:
Updated 🤓